python/graphstorm/gconstruct/transform.py

```diff
@@ -280,6 +284,32 @@ def as_out_dtype(self, feats):
         else:
             return feats.astype(self.out_dtype)
 
+    def feat2numerical(self, feats):
+        """ Cast a feature into a numerical numpy array if it is not.
+        If it is already a numerical numpy array, do nothing.
+
+        Parameters
+        ----------
+        feats: np.array
+            Input feature.
+
+        Return
+        ------
+        np.array: the cast feature.
+        """
+        if not (np.issubdtype(feats.dtype, np.integer) \
+            or np.issubdtype(feats.dtype, np.floating)):
```

Comment:
Maybe we can use https://numpy.org/devdocs/reference/arrays.scalars.html#numpy.number

Reply:
I used number before, but it has some problems, so I chose to use np.integer and np.floating.